### PR TITLE
CON-3411 Provider with new permission creates a vacancy

### PR DESCRIPTION
**Pull request by Howesy77**

### CODE REVIEW

File: src/Provider/Provider.Web/Controllers/ReviewedController.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        private readonly ReviewedOrchestrator _orchestrator;
+
+        public ReviewedController(ReviewedOrchestrator orchestrator)
+        {
+            _orchestrator = orchestrator;
+        }
+
+        [HttpGet("reviewed", Name = RouteNames.Reviewed_Index_Get)]
+        public async Task<IActionResult> Confirmation(VacancyRouteModel vrm)
+
```
Comment (cofaulco) [inline src/Provider/Provider.Web/Controllers/ReviewedController.cs:22]: blank line

File: src/Provider/Provider.Web/Orchestrators/VacancyPreviewOrchestrator.cs [hunk 2]
```diff
@@ -119,10 +127,18 @@ private async Task<SubmitVacancyResponse> SubmitActionAsync(Vacancy vacancy, Vac
...
             {
-                var command = new SubmitVacancyCommand(vacancy.Id, user, OwnerType.Provider);
-
-                await _messaging.SendCommandAsync(command);
-                response.IsSubmitted = true;
+                if (hasProviderReviewPermission)
+                {
+                    var command = new ReviewVacancyCommand(vacancy.Id, user, OwnerType.Provider);
+                    await _messaging.SendCommandAsync(command);
+                    response.IsReviewed = true;
```
Comment (cofaulco) [inline src/Provider/Provider.Web/Orchestrators/VacancyPreviewOrchestrator.cs:134]: should this property be `IsSentForReview`?

File: src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/ReviewVacancyCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+            if(vacancy.CanSubmit == false)
+                throw new InvalidOperationException(string.Format(InvalidStateExceptionMessageFormat, vacancy.Id, vacancy.Status));
+
+            if(vacancy.OwnerType != message.SubmissionOwner)
+                throw new InvalidOperationException(string.Format(InvalidOwnerExceptionMessageFormat, vacancy.Id, message.SubmissionOwner, vacancy.OwnerType));
+
+            var now = _timeProvider.Now;
+
+            if(!string.IsNullOrEmpty(message.EmployerDescription))
```
Comment (cofaulco) [inline src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/ReviewVacancyCommandHandler.cs:61]: is there any need for this check? as not setting it is the same as setting it to nullorempty

File: src/Shared/Recruit.Vacancies.Client/Application/Commands/ReviewVacancyCommand.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    public class ReviewVacancyCommand : ICommand, IRequest
+    {
+        public ReviewVacancyCommand(Guid vacancyId, VacancyUser user, OwnerType submissionOwner)
+        {
+            VacancyId = vacancyId;
+            User = user;
+            SubmissionOwner = submissionOwner;
+        }
+
+        public ReviewVacancyCommand(Guid vacancyId, VacancyUser user, string employerDescription, OwnerType submissionOwner)
```
Comment (cofaulco) [inline src/Shared/Recruit.Vacancies.Client/Application/Commands/ReviewVacancyCommand.cs:17]: employerDescription could be an optional param as opposed to 2 constructors?

**Review by cofaulco — CHANGES_REQUESTED** [review]
A few minor comments and discussed moving hardcoded operation strings to a constants file

File: src/Provider/Provider.Web/Controllers/ReviewedController.cs (hunk 1 above)
Comment (Howesy77) [reply src/Provider/Provider.Web/Controllers/ReviewedController.cs:22]: Done

File: src/Provider/Provider.Web/Orchestrators/VacancyPreviewOrchestrator.cs (hunk 2 above)
Comment (Howesy77) [reply src/Provider/Provider.Web/Orchestrators/VacancyPreviewOrchestrator.cs:134]: Done

File: src/Shared/Recruit.Vacancies.Client/Application/Commands/ReviewVacancyCommand.cs (hunk 4 above)
Comment (Howesy77) [reply src/Shared/Recruit.Vacancies.Client/Application/Commands/ReviewVacancyCommand.cs:17]: Done

File: src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/ReviewVacancyCommandHandler.cs (hunk 3 above)
Comment (Howesy77) [reply src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/ReviewVacancyCommandHandler.cs:61]: Yeah because it could've been previously set

**Review by cofaulco — APPROVED** [review]

**Pull request closed without merging**